pkg/operation/botanist/component/kubeapiserver/horizontalpodautoscaler.go

```diff
@@ -33,48 +36,96 @@ const (
 	hpaTargetAverageUtilizationMemory int32 = 80
 )
 
-// TODO: Switch to autoscaling/v2 for v1.23+ clusters as soon as we have revendored the k8s.io/* libraries with
-//
-//	v0.23.0 or above.
-func (k *kubeAPIServer) emptyHorizontalPodAutoscaler() *autoscalingv2beta1.HorizontalPodAutoscaler {
-	return &autoscalingv2beta1.HorizontalPodAutoscaler{ObjectMeta: metav1.ObjectMeta{Name: v1beta1constants.DeploymentNameKubeAPIServer, Namespace: k.namespace}}
+func (k *kubeAPIServer) emptyHorizontalPodAutoscaler() client.Object {
+	hpaObjectMeta := metav1.ObjectMeta{
+		Name:      v1beta1constants.DeploymentNameKubeAPIServer,
+		Namespace: k.namespace,
+	}
+
+	if version.ConstraintK8sGreaterEqual123.Check(k.values.Version) {
+		return &autoscalingv2.HorizontalPodAutoscaler{
+			ObjectMeta: hpaObjectMeta,
+		}
+	}
+	return &autoscalingv2beta1.HorizontalPodAutoscaler{
+		ObjectMeta: hpaObjectMeta,
+	}
```

Comment:
Do we need to do this if-else with HPAs with different contents? See also the bit below in `reconcileHorizontalPodAutoscaler`. Picking `v2beta2` would get rid of this and have the same structure/content in all HPAs.
It's true that we would need to touch this again when `v2beta2` will no longer be served (in 1.26, maybe?), but then 1.22 wouldn't need to be officially supported by us anymore, right?

Reply:
> but then 1.22 wouldn't need to be officially supported by us anymore, right?

What do you mean?

Reply:
<img width="874" alt="Screenshot 2022-09-12 at 09 20 32" src="https://user-images.githubusercontent.com/2256887/189595477-d1185f92-af43-4249-a2f9-a38f4c9fc39f.png">

k8s 1.22 has EOL end of October, so when we need to migrate off v2beta2, the smallest supported version would be 1.23 – which has v2 available.

I was thinking on how to avoid the if-else stuff for all things HVPA/HPA, not sure if that makes sense.

Reply:
I don't think the Gardener project strictly follows the same EOL lifecycle for supported Kubernetes versions. Typically, Gardener supports older Kubernetes versions for a longer time (longer than the community's EOL).
For now, we have to live with the if-else constructs until we finally decide to drop support for old versions.